e3sm_diags/driver/utils/regrid.py

```diff
@@ -91,8 +91,8 @@ def subset_and_align_datasets(
 
     # Subset on a specific region.
     if "global" not in region:
-        ds_test_new = _subset_on_region(ds_test, var_key, region)
-        ds_ref_new = _subset_on_region(ds_ref, var_key, region)
+        ds_test_new = _subset_on_region(ds_test_new, var_key, region)
+        ds_ref_new = _subset_on_region(ds_ref_new, var_key, region)
```

Comment:
Incorrect dataset objects were being passed to `_subset_on_region`. If land sea mask was applied, it was being ignored here and downstream.

This affected `lat_lon` and `polar`.